boa/src/object/internal_methods.rs

```diff
@@ -390,6 +392,70 @@ impl GcObject {
         }
     }
 
+    /// Gets own property of 'Object'
+    ///
+    #[inline]
+    pub fn get_own_property(&self, key: &PropertyKey) -> Option<PropertyDescriptor> {
+        let _timer = BoaProfiler::global().start_event("Object::get_own_property", "object");
+
+        let object = self.borrow();
+        match object.data {
+            ObjectData::String(_) => self.string_exotic_get_own_property(key),
+            _ => self.ordinary_get_own_property(key),
+        }
+    }
+
+    /// StringGetOwnProperty abstract operation
+    ///
+    /// More information:
+    ///  - [ECMAScript reference][spec]
+    ///
+    /// [spec]: https://tc39.es/ecma262/#sec-stringgetownproperty
+    #[inline]
+    pub fn string_get_own_property(&self, key: &PropertyKey) -> Option<PropertyDescriptor> {
+        let object = self.borrow();
+
+        match key {
+            PropertyKey::Index(index) => {
+                let string = object.as_string().unwrap();
+                let pos = *index as usize;
+
+                if pos >= string.len() {
+                    return None;
+                }
+
+                let char = if let Some(utf16_val) = string.encode_utf16().nth(pos) {
```

Comment:
I don't like using the type's name, `char`, as a variable name, GH's parser in particular considers it a keyword, I recommend the spec's `resultStr`.